packages/ui-components/src/components/ComboBox/ComboBox.component.tsx

```diff
@@ -429,59 +450,64 @@ export const ComboBox = ({
   )
 }
 
-ComboBox.propTypes = {
+export type ComboBoxWidth = "full" | "auto"
+
+//eslint-disable-next-line no-unused-vars
+type OnChangeHandler = (value: string) => void
+
+export interface ComboBoxProps {
   /** The aria-label of the ComboBox. Defaults to the label if label was passed. */
-  ariaLabel: PropTypes.string,
+  ariaLabel?: string
   /** The children to Render. Use `ComboBox.Option` elements. */
-  children: PropTypes.node,
+  children?: React.ReactElement<ComboBoxOptionProps> | React.ReactElement<ComboBoxOptionProps>[] | null
```

Comment:
**(Note for Future Improvements)**
We usually just define this as `React.ReactNode`. Makes sense to restrict to only `ComboBoxOptionProps` though. Seeing this example, I wonder if we could make this improvement in other components too, where we expect a specific element as children?
CC: @andypf @barsukov

Reply:
I also think it is make sense in this case but we should be careful regarding other components. But sounds reasonable for me tbh 👍

Reply:
I changed it back to ReactNode to avoid other app breaking. I will add it to improvements though as it seems reasonable type restriction.

Reply:
Added to improvements list. Resolving this thread.